File: 2017/solution.py
class Solution:
    def gridGame(self, grid: list[list[int]]) -> int:
        ps = grid.copy()
        ROWS, COLS = 2, len(grid[0])
        for i in range(1, COLS):
            ps[0][i] += ps[0][i - 1]
            ps[1][i] += ps[1][i - 1]

        res = ps[0][-1] - ps[0][0]

        for i in range(1, COLS):
            top = ps[0][-1] - ps[0][i]
            bottom = ps[1][i - 1]
            res = min(res, max(top, bottom))
        return res

File: 2017/solution.go
package main

import "math"

func gridGame(grid [][]int) int64 {
	totals := [2]int64{0, 0}
	COLS := len(grid[0])
	for _, v := range grid[0] {
		totals[0] += int64(v)
	}
	var res int64 = math.MaxInt64
	for i := 0; i < COLS; i++ {
		totals[0] -= int64(grid[0][i])
		res = min(res, max(totals[0], totals[1]))
		totals[1] += int64(grid[1][i])
	}
	return res
}
